framework/src/base/FEProblemBase.C

```diff
@@ -129,6 +129,9 @@ validParams<FEProblemBase>()
                         false,
                         "This causes PETSc to error if it had to reallocate memory in the Jacobian "
                         "matrix due to not having enough nonzeros");
+  params.addParam<bool>("ignore_zeros_in_jacobian",
+                        false,
+                        "This stops zero values from creating a zero location in the matrix ");
```

Comment:
Do not explicitly store zero values in  the Jacobian matrix if ` ignore_zeros_in_jacobian = true`